test/unit/stream_alert_rule_processor/test_handler.py

```diff
@@ -0,0 +1,231 @@
+'''
+Copyright 2017-present, Airbnb Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+'''
+import base64
+import logging
+
+from mock import call, Mock, mock_open, patch
+
+from nose.tools import (
+    assert_equal,
+    assert_false,
+    assert_list_equal,
+    assert_true,
+    raises
+)
+
+from stream_alert.rule_processor import LOGGER
+from stream_alert.rule_processor.config import ConfigError
+from stream_alert.rule_processor.handler import StreamAlert
+from stream_alert.rule_processor.handler import load_config
+
+
+from unit.stream_alert_rule_processor.test_helpers import (
+    _get_mock_context,
+    _get_valid_event
+)
+
+
+class TestStreamAlert(object):
+    """Test class for StreamAlert class"""
+    @classmethod
+    @patch('stream_alert.rule_processor.handler.load_config',
+           lambda: load_config('test/unit/conf/'))
+    def setup_class(cls):
+        """Setup the class before any methods"""
+        cls.__sa_handler = StreamAlert(_get_mock_context(), False)
+
+    @classmethod
+    def teardown_class(cls):
+        """Teardown the class after all methods"""
+        cls.__sa_handler = None
+
+    def teardown(self):
+        """Teardown the class after each methods"""
+        del self.__sa_handler._alerts[:]
+        self.__sa_handler._failed_record_count = 0
+        self.__sa_handler.env['lambda_alias'] = 'development'
+        self.__sa_handler.enable_alert_processor = False
```

Comment:
Again, all of this teardown suggests that the `StreamAlert` should be built for each test (in `setup`), not for the whole class (`setup_class`). The extra initialization cost is minimal compared to the benefit of guaranteeing a consistent environment for testing.

TL;DR avoid `setup_class` unless you absolutely need state saved across multiple test runs.